File: 2274-keep-multiplying-found-values-by-two/keep-multiplying-found-values-by-two.js
/**
 * @param {number[]} nums
 * @param {number} original
 * @return {number}
 */
var findFinalValue = function(nums, original) {
    for(let i = 0; i < nums.length; i++){
        if(nums[i] == original){
            return findFinalValue(nums, original*2);
        }
       }
       return original;
};

File: 2274-keep-multiplying-found-values-by-two/keep-multiplying-found-values-by-two.java
class Solution {
    public int findFinalValue(int[] nums, int original) {
        HashSet<Integer> hs = new HashSet<>();
        for(int i : nums) hs.add(i);
        for (int i : nums) {
            if (hs.contains(original))
                original = original * 2;
        }
        return original;
    }
}